src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -296,23 +297,30 @@ static constexpr const auto kLibSSLMatchers = MakeArray<SSLLibMatcher>({
         .libssl = kLibSSL_1_1,
         .libcrypto = "libcrypto.so.1.1",
         .search_type = HostPathForPIDPathSearchType::kSearchTypeEndsWith,
+        .probe_suffix = "_syscall_fd_access",
     },
     SSLLibMatcher{
         .libssl = kLibSSL_3,
         .libcrypto = "libcrypto.so.3",
         .search_type = HostPathForPIDPathSearchType::kSearchTypeEndsWith,
+        .probe_suffix = "_syscall_fd_access",
     },
     SSLLibMatcher{
         // This must match independent of python version and INSTSONAME suffix
         // (e.g. libpython3.10.so.0.1).
         .libssl = kLibPython,
         .libcrypto = kLibPython,
         .search_type = HostPathForPIDPathSearchType::kSearchTypeContains,
+        .probe_suffix = "_syscall_fd_access",
     },
+    // non BIO native TLS applications cannot be probed by accessing the socket fd
+    // within the underlying syscall. Specifying an empty probe_suffix results in TLS
+    // tracing via the legacy mechanism (finding socket fd with user space memory offsets).
     SSLLibMatcher{
```

Comment:
The concept of a `probe_suffix` is not self documenting and although there is a comment here (enough of a clue to go on) there may be a better way to directly describe what is going on here. Possibly something like this:
1. An struct member (enum perhaps) that identifies the SSL technique (e.g. nested, legacy).
2. one of the following:
... a. Function that returns the probe fn. name based on the relevant parameters.
... b. A map or array that stores the probe fn. suffix based on the enum.

Reply:
I've refactored this to use 1). Let me know what you think.